go-runtime/compile/compile_integration_test.go

```diff
@@ -15,9 +15,8 @@ func TestNonExportedDecls(t *testing.T) {
 		in.CopyModule("echo"),
 		in.Deploy("echo"),
 		in.CopyModule("notexportedverb"),
-		in.ExpectError(
-			in.ExecWithOutput("ftl", "deploy", "notexportedverb"),
-			"call first argument must be a function but is an unresolved reference to echo.Echo, does it need to be exported?"),
+		in.ExecWithExpectedError("call first argument must be a function but is an unresolved reference to echo.Echo, does it need to be exported?",
```

Comment:
Why the change? The previous function was generalisable to any Action.

Reply:
I went in circles a bit on this:
- When I used `ExecWithOutput` I expected to be able to use the output. It didn't indicate that it was only capturing the output in case of an error
- I then tried to make `ExecWithOutput` work for both success success and error cases:
    - For success cases, call a func with the output (like the replacement `ExecWithOutput` action I added)
    - For error cases throw the error with the output (old behaviour)
- It then felt a bit messy for the error case so I tried cleaning it up.

I can go back to `ExecWithOutput` handling both cases. Thoughts?

Reply:
I think it's fine to introduce new functionality, but that doesn't seem like a reason to remove `in.ExpectError()`?

Reply:
Yeh I see what you mean. I've moved them back into one func called `ExecWithOutput` which works like before for the error case (used in conjunction with `ExpectError` and an empty capture func).